src/metrics/mod.rs

```diff
@@ -516,10 +524,10 @@ pub async fn middleware(
                 "remote_addr": conn_info.remote_addr.ip().to_string(),
                 "request_time": duration_full.as_secs_f64(),
                 "request_time_headers": 0,
-                "cache_status": meta.cache_status,
+                "cache_status": resp_meta.cache_status,
                 "cache_status_nginx": cache_status_str,
-                "cache_bypass_reason": meta.cache_bypass_reason,
-                "upstream": "127.0.0.1",
+                "cache_bypass_reason": resp_meta.cache_bypass_reason,
+                "upstream": req_meta.backend.unwrap_or_default(),
```

Comment:
Should this also be updated in the comment above (lines 449 - 489). Unfortunately I cannot comment on unchanged lines.

Reply:
I'll remove the commented lines for now to make code clear. Later I can re-create them or dig out of git if need be.